src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs

```diff
@@ -63,31 +65,51 @@ public HttpRequestMessage BuildRequest(FormulaValue[] args)
             // Header names are not case sensitive.
             // From RFC 2616 - "Hypertext Transfer Protocol -- HTTP/1.1", Section 4.2, "Message Headers"
             var headers = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            string body = null;
 
             var map = _argMapper.ConvertToSwagger(args);
             foreach (var param in _argMapper._openApiParameters)
             {
-                if (map.TryGetValue(param.Name, out var paramValue))
+                if (param.In == FxParameterLocation.Body && param.Schema.Properties != null && param.Schema.Properties.Any())
+                {
+                    var props = new Dictionary<string, object>();
+
+                    foreach (var property in param.Schema.Properties)
+                    {
+                        if (map.TryGetValue(property.Key, out var paramValue))
+                        {
+                            props.Add(property.Key, JsonSerializer.Serialize(paramValue.ToObject()));
+                        }
+                    }
+
+                    body = $"{{ {string.Join(", ", props.Select(p => $"{p.Key}: {p.Value}"))} }}";
```

Comment:
>body = $"{{ {string.Join(", ", props.Select(p => $"{p.Key}: {p.Value}"))} }}"; [](http://example.com/codeflow?start=20&length=78)

??
Is this attempting Json Serialization?

Reply:
I could try serializing a dictionary instead, that will be cleaner